### PR TITLE
Add extra volumes and volume mounts on scheduler/workers

### PR DESCRIPTION
**Pull request by jedcunningham**
Add support for extra volumes and volume mounts in scheduler, celery workers, and kubernetes executor workers.

## 🎟 Issue(s)

Resolves astronomer/issues#2684
Resolves astronomer/issues#2685 (in coordination with #192)

Co-authored-by: Daniel Imberman <daniel.imberman@gmail.com>

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

File: tests/pod-template-file_test.yaml [hunk 1]
```diff
@@ -19,3 +19,23 @@ tests:
...
+      - contains:
+          path: spec.volumes
+          content:
+            name: test-volume
+            emptyDir: {}
+      - contains:
+          path: spec.containers[0].volumeMounts
+          content:
+            name: test-volume
+            mountPath: /opt/test
```
Comment (ianstanton) [inline tests/pod-template-file_test.yaml:41]:
```suggestion
            mountPath: /opt/test

```

File: tests/scheduler_scheduler-deployment_test.yaml [hunk 2]
```diff
@@ -35,3 +35,23 @@ tests:
...
+      - contains:
+          path: spec.template.spec.volumes
+          content:
+            name: test-volume
+            emptyDir: {}
+      - contains:
+          path: spec.template.spec.containers[0].volumeMounts
+          content:
+            name: test-volume
+            mountPath: /opt/test
```
Comment (ianstanton) [inline tests/scheduler_scheduler-deployment_test.yaml:57]:
```suggestion
            mountPath: /opt/test

```

File: tests/workers_worker-deployment_test.yaml [hunk 3]
```diff
@@ -42,3 +41,24 @@ tests:
...
+      - contains:
+          path: spec.template.spec.volumes
+          content:
+            name: test-volume
+            emptyDir: {}
+      - contains:
+          path: spec.template.spec.containers[0].volumeMounts
+          content:
+            name: test-volume
+            mountPath: /opt/test
```
Comment (ianstanton) [inline tests/workers_worker-deployment_test.yaml:64]:
```suggestion
            mountPath: /opt/test

```

**Review by ianstanton — CHANGES_REQUESTED** [review]
Some nitpicks

**Review by ianstanton — APPROVED** [review]

**Comment by jedcunningham** [comment]
Take another look @kaxil @ianstanton, I've added `extraObjects`.

**Review by ianstanton — APPROVED** [review]

File: templates/extra-objects.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{{ range .Values.extraObjects }}
+---
+{{ tpl (toYaml .) $ }}
+{{ end }}
```
[the remark this answers is not in the document]
Comment (jedcunningham) [reply templates/extra-objects.yaml:4]: I think it covers the common cases. I'll link out to this in the docs.

**Review by kaxil — APPROVED** [review]

**Pull request merged**